server/shared/src/main/scala/org/http4s/server/middleware/CORS.scala

```diff
@@ -194,10 +194,11 @@ object CORS {
     "Depends on a deficient `CORSConfig`. See https://github.com/http4s/http4s/security/advisories/GHSA-52cf-226f-rhr6. If config.anyOrigin is true and config.allowCredentials is true, then the `Access-Control-Allow-Credentials` header will be suppressed starting with 0.22.3.",
     "0.21.27",
   )
-  @nowarn("cat=deprecation")
+  @nowarn // silence deprecation warning on Scala 3
   def apply[F[_], G[_]](http: Http[F, G], config: CORSConfig = CORSConfig.default)(implicit
       F: Applicative[F]
   ): Http[F, G] = {
+    val purposelyUnused = 0 // to satisfy @nowarn on Scala 2
```

Comment:
```suggestion
    def purposelyUnused = 0 // to satisfy @nowarn on Scala 2
```

Reply:
Good suggestion :) I was curious about the bytecode difference.

Interestingly enough, the `val` generates an unused `boolean` (?).

```java
    public <F, G> Kleisli<F, Request<G>, Response<G>> apply(Kleisli<F, Request<G>, Response<G>> http, CORSConfig config, Applicative<F> F) {
        boolean purposelyUnused = false;
```

Meanwhile the `def` generates an unused private static method.

```java
    private static final int purposelyUnused$1() {
        return 0;
    }
```

The `def` is better, because it doesn't change the method definition.